### ISSUE TITLE
[Bug] `v2/universe/names/` sometimes returns "error": "Failed to read request body"

### ISSUE THREAD

**Issue by exodus4d** [ccpgames/esi-issues#861]
This endpoint returns an error *sometimes* so it _might_ be a bug in CCPs side.... 
I send the **same** request **twice in a row**,  with same request params and **sometimes**  I get this error response: `[body] => {"error": "Failed to read request body"}` . 

I send *199* Ids to that endpoint and expect *24 alliance names* and *175 character names*. 
You can see that request params are totally equal for both requests, nevertheless the second requests returns that error:

### This works:
```
Array
(
    [timeout] => 4
    [method] => POST
    [user_agent] => Pathfinder DEV - v1.3.5 | https://github.com/exodus4d (PATHFINDER)
    [header] => Array
        (
            [0] => Content-Type: application/json
        )
    [content] => [99003144,99005462,99006394,99006805,99007920,99002410,99002957,99002964,99003109,99006812,99007415,99008147,99000448,473668485,99001327,99006343,99007983,99008210,99000906,99006225,99006334,99006958,99008129,99008232,802597327,812769358,806966246,799787049,799809637,813002867,798410068,798186130,813519168,804125344,877400860,861452033,870660211,866597064,832142555,826549293,944366752,908832867,925627594,917148146,928439395,913565630,891970360,896612723,887929302,890076366,949296436,952897385,961036810,965549954,972766515,973848052,982584902,982161749,967333435,961402249,969693743,976742430,1051993578,1017617892,1014347414,1021924274,1027954267,1011321806,1040119491,1040560303,1054716731,1053660463,1241860972,1243255020,1245784934,1238247230,1236989895,1221948374,1210042884,1207043762,1209166500,1222861935,1225029383,1218361075,1170241755,1187341687,1166344763,1153516880,1175724137,1176009870,1177815945,1165054482,1173054082,1252309285,1347332575,1350764604,1322966933,1265501626,1327285925,1292009733,1276346482,1267257668,1276503436,1316855186,1297165156,1315612744,1298639839,1296818051,1277553127,1284639027,1260338877,1281543116,1283204734,1280518888,1293567696,1364446188,1359210441,1403183448,1369325662,1375073825,1419279328,1405418131,1401615998,1442012977,1428721891,1442317275,1415957058,1414713509,1373459272,1345158944,1356945472,1444393816,1497548730,1488203591,1489692013,1470028677,1483928548,1485238070,1501406986,1497822879,1461659569,1445905918,1520302915,1072795574,1073256735,1054828882,1078597891,1082398624,1062897087,1104890437,1894531169,1568211092,1578620265,1572599291,1794237602,1825206774,1840621219,1857050901,1855358253,1552453603,1537860381,1532457169,1555374790,1556571928,1828673046,1833511119,1819300754,1830952778,1600278891,1585059630,1582485839,1582627746,1608992781,1605863611,1829171350,1884145283,1885508352,1878948626,1860851964,1811086512,1824547123,1949142540,1997253145,2007811563,2004511421,2009868427,1991926078,2015099036,1967069961,1982624581,1965443374,1963012437,1963540231,1946869415,1925018023,1930460366,1945595704,1943935055,1910043459]
)
Array
(
    [request] => Array
        (
            [0] => POST https://esi.tech.ccp.is/v2/universe/names/?datasource=tranquility
            [1] => Content-Type: application/json
            [2] => Host: esi.tech.ccp.is
            [3] => Accept-Encoding: gzip,deflate
            [4] => User-Agent: Pathfinder DEV - v1.3.5 | https://github.com/exodus4d (PATHFINDER)
            [5] => Connection: close
            [6] => Content-Length: 2105
        )
    [body] => [{"id":99003144,"name":"Scary Wormhole People","category":"alliance"},{"id":99005462,"name":"WORMHOLE PVP CHANNEL","category":"alliance"}, ...MORE DATA...]
    [headers] => Array
        (
            [0] => HTTP/1.1 100 Continue
            [1] => HTTP/1.1 200 OK
            [2] => Access-Control-Allow-Credentials: true
            [3] => Access-Control-Allow-Headers: Content-Type,Authorization,X-User-Agent
            [4] => Access-Control-Allow-Methods: POST,OPTIONS
            [5] => Access-Control-Allow-Origin: *
            [6] => Access-Control-Expose-Headers: Content-Type,Warning,X-Pages,X-ESI-Error-Limit-Remain,X-ESI-Error-Limit-Reset
            [7] => Access-Control-Max-Age: 600
            [8] => Allow: OPTIONS,POST
            [9] => Content-Encoding: gzip
            [10] => Content-Type: application/json; charset=UTF-8
            [11] => Date: Sun, 22 Apr 2018 17:36:34 GMT
            [12] => Strict-Transport-Security: max-age=31536000
            [13] => Vary: Accept-Encoding
            [14] => X-Esi-Ab-Test: 1
            [15] => X-Esi-Error-Limit-Remain: 99
            [16] => X-Esi-Error-Limit-Reset: 26
            [17] => X-Esi-Request-Id: 212e564b-5298-4391-8332-4b7e42da77c5
            [18] => Transfer-Encoding: chunked
            [19] => Via: 1.1 google
            [20] => Alt-Svc: clear
            [21] => Connection: close
        )
    [engine] => cURL
    [cached] => 
    [error] => 
)
```

### This Fails:
```
Array
(
    [timeout] => 4
    [method] => POST
    [user_agent] => Pathfinder DEV - v1.3.5 | https://github.com/exodus4d (PATHFINDER)
    [header] => Array
        (
            [0] => Content-Type: application/json
        )
    [content] => [99003144,99005462,99006394,99006805,99007920,99002410,99002957,99002964,99003109,99006812,99007415,99008147,99000448,473668485,99001327,99006343,99007983,99008210,99000906,99006225,99006334,99006958,99008129,99008232,802597327,812769358,806966246,799787049,799809637,813002867,798410068,798186130,813519168,804125344,877400860,861452033,870660211,866597064,832142555,826549293,944366752,908832867,925627594,917148146,928439395,913565630,891970360,896612723,887929302,890076366,949296436,952897385,961036810,965549954,972766515,973848052,982584902,982161749,967333435,961402249,969693743,976742430,1051993578,1017617892,1014347414,1021924274,1027954267,1011321806,1040119491,1040560303,1054716731,1053660463,1241860972,1243255020,1245784934,1238247230,1236989895,1221948374,1210042884,1207043762,1209166500,1222861935,1225029383,1218361075,1170241755,1187341687,1166344763,1153516880,1175724137,1176009870,1177815945,1165054482,1173054082,1252309285,1347332575,1350764604,1322966933,1265501626,1327285925,1292009733,1276346482,1267257668,1276503436,1316855186,1297165156,1315612744,1298639839,1296818051,1277553127,1284639027,1260338877,1281543116,1283204734,1280518888,1293567696,1364446188,1359210441,1403183448,1369325662,1375073825,1419279328,1405418131,1401615998,1442012977,1428721891,1442317275,1415957058,1414713509,1373459272,1345158944,1356945472,1444393816,1497548730,1488203591,1489692013,1470028677,1483928548,1485238070,1501406986,1497822879,1461659569,1445905918,1520302915,1072795574,1073256735,1054828882,1078597891,1082398624,1062897087,1104890437,1894531169,1568211092,1578620265,1572599291,1794237602,1825206774,1840621219,1857050901,1855358253,1552453603,1537860381,1532457169,1555374790,1556571928,1828673046,1833511119,1819300754,1830952778,1600278891,1585059630,1582485839,1582627746,1608992781,1605863611,1829171350,1884145283,1885508352,1878948626,1860851964,1811086512,1824547123,1949142540,1997253145,2007811563,2004511421,2009868427,1991926078,2015099036,1967069961,1982624581,1965443374,1963012437,1963540231,1946869415,1925018023,1930460366,1945595704,1943935055,1910043459]
)
Array
(
    [request] => Array
        (
            [0] => POST https://esi.tech.ccp.is/v2/universe/names/?datasource=tranquility
            [1] => Content-Type: application/json
            [2] => Host: esi.tech.ccp.is
            [3] => Accept-Encoding: gzip,deflate
            [4] => User-Agent: Pathfinder DEV - v1.3.5 | https://github.com/exodus4d (PATHFINDER)
            [5] => Connection: close
            [6] => Content-Length: 2105
        )
    [body] => {"error": "Failed to read request body"}
    [headers] => Array
        (
            [0] => HTTP/1.1 100 Continue
            [1] => HTTP/1.1 500 Internal Server Error
            [2] => Access-Control-Allow-Credentials: true
            [3] => Access-Control-Allow-Headers: Content-Type,Authorization,X-User-Agent
            [4] => Access-Control-Allow-Methods: POST,OPTIONS
            [5] => Access-Control-Allow-Origin: *
            [6] => Access-Control-Expose-Headers: Content-Type,Warning,X-Pages,X-ESI-Error-Limit-Remain,X-ESI-Error-Limit-Reset
            [7] => Access-Control-Max-Age: 600
            [8] => Content-Length: 40
            [9] => Content-Type: application/json
            [10] => Date: Sun, 22 Apr 2018 17:35:13 GMT
            [11] => Strict-Transport-Security: max-age=31536000
            [12] => X-Esi-Ab-Test: 1
            [13] => X-Esi-Error-Limit-Remain: 100
            [14] => X-Esi-Error-Limit-Reset: 47
            [15] => Via: 1.1 google
            [16] => Alt-Svc: clear
            [17] => Connection: close
        )
    [engine] => cURL
    [cached] => 
    [error] => 
)
```

**Comment by exodus4d** [comment]
Oh it looks like I figured it out :slightly_smiling_face: As you see I always get those response headers:
```
            [0] => HTTP/1.1 100 Continue
            [1] => HTTP/1.1 500 Internal Server Error
```

1. The first "problem" was a bug in my Header "parsing script" looking for the HTTP response status.... my regRex always returns the first match: `100`, Request was processed as "success" and no _retry loop_ starts -_-
2. The core problem still exists after that -> "error response" :slightly_smiling_face: This is because cURL never sends the request body.... 
Explanation:
1. cURL *automatically* adds the `Expect: 100-continue` header to **POST* requests *> 1024byte* 
2. This allows the server to **reject** or **accept** the following **large** request body (which will be send afterwards)
3. CCP can´t deal with this `Expect: 100-continue` and rejects the request with *500* because they can´t find the body in the initial request

How to fix:
I had to add add an **empty** `'Expect:'` so cURL does not overwrite/add the default `Expect: 100-continue`

See: https://gms.tf/when-curl-sends-100-continue.html
